### PR TITLE
Cluster provisioning scripts for integration tests in api-gateway repo

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Cluster provisioning scripts for integration tests in api-gateway repo

**Related issue(s)**

https://github.com/kyma-project/api-gateway/issues/213

### CODE REVIEW

File: tests/integration/scripts/gardener/gcp.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    fi
+
+    CLEANUP_CLUSTER="true"
+      # enable trap to catch kyma provision failures
+      trap gardener::reprovision_cluster ERR
+      # decreasing attempts to 2 because we will try to create new cluster from scratch on exit code other than 0
+      kyma provision gardener gcp \
+        --secret "${GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME}" \
+        --name "${CLUSTER_NAME}" \
+        --project "${GARDENER_KYMA_PROW_PROJECT_NAME}" \
```
Comment (barchw) [inline tests/integration/scripts/gardener/gcp.sh:96]: Maybe we could use goat project for easier access to the clusters and to make sure they are deleted?

File: tests/integration/scripts/integration-gardener.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
+
+if [[ "${HIBERNATION_ENABLED}" == "true" ]]; then
+    gardener::hibernate_kyma
+    sleep 120
+    gardener::wake_up_kyma
+fi
+
+"${API_GATEWAY_SOURCES_DIR}/tests/integration/scripts/jobguard/run.sh"
+api-gateway::launch_integration_tests
```
Comment (barchw) [inline tests/integration/scripts/integration-gardener.sh:109]: Can we use our Makefile instead?

**Review by barchw — CHANGES_REQUESTED** [review]

File: tests/integration/scripts/gardener/gcp.sh (hunk 1 above)
Comment (videlov) [reply tests/integration/scripts/gardener/gcp.sh:96]: @kasiakepka do we have creds for our goat gardener project within Prow as preset or similar?

Comment (videlov) [reply tests/integration/scripts/gardener/gcp.sh:96]: Does Neighbors have cluster leftover cleanup scripts in place? I would not insist of having it in our project, unless it is really necessary.

File: tests/integration/scripts/integration-gardener.sh (hunk 2 above)
Comment (barchw) [reply tests/integration/scripts/integration-gardener.sh:109]: That's fine 👍

File: tests/integration/scripts/gardener/gcp.sh (hunk 1 above)
Comment (kasiakepka) [reply tests/integration/scripts/gardener/gcp.sh:96]: Yes, by default each script provisions and deprovisions the cluster but there is a possibility to override that.

Comment (videlov) [reply tests/integration/scripts/gardener/gcp.sh:96]: We have this in our script, the question if our script somehow miss to deprovision a cluster, is there a cleaner procedure/script/job of some art running by neighbours team that cleans some leftovers? or do someone monitor that?

Comment (kasiakepka) [reply tests/integration/scripts/gardener/gcp.sh:96]: There is gardener cleanup function https://github.com/kyma-project/test-infra/blob/6f9b3cde61c0f00fad42ddd29d71aac537e3f73c/prow/scripts/lib/gardener/gcp.sh#L20
@videlov I am not sure if I understand you correctly :) give me a call if you need to clarify things.

**Review by barchw — APPROVED** [review]

**Pull request merged**